[CORL-1747] Live Chat: Merge the root and conversation "comment entered" subscriptions

nick-funk opened the pull request:
## What does this PR do?

Merges the live chat root and conversation "comment entered" subscriptions together to prevent race conditions

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## How do I test this PR?

- Use live chat
- Create comments
- See that it updates the "jump to new", live insertion, etc on both the main chat and also the conversation views

nick-funk committed:
Merge the two comment entered subscriptions into one

nick-funk committed:
Delete old reply subscription, it is no longer used

alinhle commented on src/core/client/stream/tabs/Live/LiveChatContainer.tsx:
```diff
@@ -133,7 +134,7 @@ const LiveChatContainer: FunctionComponent<Props> = ({
     showConversation,
     hideConversation,
-  ] = useConversation(eventEmitter);
+  ] = useConversation(eventEmitter, setLocal);
```
Instead of passing `setLocal` here which is initialized with a lot more than just conversation related data, do a `useLocal` inside of `useConversation` and return the `conversationRootCommentID`.

alinhle commented on src/core/client/stream/tabs/Live/LiveCommentEnteredSubscription.tsx:
```diff
@@ -13,13 +13,26 @@ import { LiveCommentEnteredSubscription } from "coral-stream/__generated__/LiveC
+
+  return liveChat.conversationRootCommentID;
+}
+
```
What do you think about calling these `isLiveMainStreamInsertionEnabled`, `isLiveConversationInsertionEnabled`, `getConversationRootCommentID` instead? I mistook them for variables on first glance instead of functions.

alinhle reviewed (changes requested):
Pretty awesome! Thanks. I have only minor requests.

nick-funk committed:
Centralize useLocal and useCoralContext within useConversation

They are no longer parameters, but are instead inherent
internal dependencies, making this hook easier to instantiate
and use.

nick-funk committed:
Clarify retrieval functions within LiveChatCommentEnteredSubscription

alinhle committed:
chore: remove unused

alinhle reviewed (approved):
Just removed an unused field in a commit, everything else looks awesome!

kodiakhq[bot] committed:
Merge branch 'feat/live-chat-epic' into feat/CORL-1747-subscription-merge
